packages/core/src/components/button/buttons.tsx

```diff
@@ -94,8 +106,8 @@ function useSharedButtonAttributes<E extends HTMLAnchorElement | HTMLButtonEleme
             [Classes.FILL]: fill,
             [Classes.LARGE]: large,
             [Classes.LOADING]: loading,
-            [Classes.MINIMAL]: minimal,
-            [Classes.OUTLINED]: outlined,
+            [Classes.MINIMAL]: minimal || variant === "minimal",
+            [Classes.OUTLINED]: outlined || variant === "outlined",
```

Comment:
This bit of code has some unintended side effects. I think that we should make the behavior be that `variant` supersedes both `outlined` and `minimal`.  However, if we do the following:

```tsx
<Button text="Button" variant="minimal" outlined={true} />
```

We end up with an outlined button instead of a minimal button as would be hoped for. The same issue occurs with `ButtonGroup`s.

Reply:
Technically, this behavior [already exists](https://codesandbox.io/p/sandbox/button-minimal-outlined-d99k9v) between the `minimal` and `outlined` props where `outlined` overwrites the styles of `minimal` if both are applied simultaneously.
```tsx
<Button text="Button" minimal={true} outlined={true} />
```

I think it is fair, however, that we wouldn't want to continue this behavior!

bb09d14212cc3501deac661a6298159eca71c587 wraps the logic into a utility function, providing the proper application of classes based on precedence. This technically changes the behavior of the presence of classes (for the case described above), but in practice it doesn't matter much since, as we've established, the `outlined` classes overwrite `minimal` anyways.